std/container/slist.d

```diff
@@ -354,15 +354,19 @@ Returns: The number of elements inserted
 
 Complexity: $(BIGOH m), where $(D m) is the length of $(D stuff)
      */
-    size_t insertFront(Stuff)(Stuff stuff)
+    size_t insertFront(Stuff)(scope Stuff stuff)
     if (isInputRange!Stuff && isImplicitlyConvertible!(ElementType!Stuff, T))
     {
+        import std.traits : hasIndirections;
         initialize();
         size_t result;
         Node * n, newRoot;
         foreach (item; stuff)
         {
-            auto newNode = new Node(null, item);
+            static if (hasIndirections!T)
+                auto newNode = new Node(null, item.dup);
```

Comment:
Ok this same thing came up in https://github.com/dlang/phobos/pull/6281 and I'm at the point where I believe this should be considered a fatal flaw of DIP1000.

How is anyone supposed to implement DIP1000 if any time they need to assign a `scope` variable to a result `struct`/`class` they need to copy it?

Am I missing an obvious solution here?

Reply:
Welcome to the club. Yes DIP1000 is about as useful as head `const`. Or probably worse.